src/plugins/slurm/slurm-spank-auks.c

```diff
@@ -501,51 +482,68 @@ spank_auks_remote_init (spank_t sp, int ac, char *av[])
 		goto exit;
 	}
 
-	/* get user credential */
-	fstatus = auks_api_get_cred(&engine,uid,auks_credcache);
-	if ( fstatus != AUKS_SUCCESS ) {
-		xerror("unable to get user %u cred : %s",uid,
-		      auks_strerror(fstatus));
-		unlink(auks_credcache);
-		auks_credcache[0]='\0';
+	/* Get auks cred */
+	fstatus = auks_api_get_auks_cred(&engine,uid,&cred);
+	if( fstatus ) {
+		xerror("unable to unpack auks cred from reply : %s",
+		       auks_strerror(fstatus));
+		fstatus = AUKS_ERROR_API_CORRUPTED_REPLY ;
 		goto unload;
```

Comment:
In case of error, you will jump to unload where you will try to free cred which is not initialized, certainly resulting in a segv or something similar